### PR TITLE
Put `mamba` under `condabin/` too

### PR DESCRIPTION
**Pull request by jaimergp**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Some users have `$PREFIX/condabin` in `PATH`, and updating to mamba 2.0 breaks their workflows because it's only putting the executable under `bin/`, which results in errors like `executable not found` etc. See https://github.com/conda-incubator/setup-miniconda/actions/runs/11083642307/job/30798086717?pr=367#step:3:906.

### CODE REVIEW

**Review by jjerphan — APPROVED** [review]
Thank you for this contribution, @jaimergp.

File: recipe/build_mamba.bat [hunk 1]
```diff
@@ -36,5 +36,8 @@ if /I "%PKG_NAME%" == "mamba" (
     cmake --build build-mamba/ --parallel %CPU_COUNT%
     if errorlevel 1 exit 1
     cmake --install build-mamba/
-
+    :: Place mamba executable copy to condabin/
+    mkdir "%PREFIX%\condabin"
+    echo ^"%LIBRARY_BIN%\mamba.exe^" %%* > "%PREFIX%\condabin\mamba.bat"
```
Comment (JohanMabille) [inline recipe/build_mamba.bat:41]: I need to check but I'm pretty sure this would break the mamba acivate / mamba deactivate features.

**Review by JohanMabille — CHANGES_REQUESTED** [review]

File: recipe/build_mamba.bat (hunk 1 above)
Comment (jaimergp) [reply recipe/build_mamba.bat:41]: We can also place a copy of the full executable there. Would that break activation?

**Comment by JohanMabille** [comment]
Back on this, these scripts should be generated via `mamba shell init -s cmd.exe --root-prefix=$MAMBA_DEFAULT_ENV` (or any shell you want to initialise) when running the miniconda installer (same for mamba forge).

**Comment by JohanMabille** [comment]
If the are paths in.bat files are replaced when relocating packages, it should work and that should be the proper fix.

**Comment by jaimergp** [comment]
The only `mamba` subcommand I can find is `mamba shell hook --shell cmd.exe %PREFIX%`. I think that should work, let's give it a try.

**Comment by jaimergp** [comment]
CI now shows this for the `mamba` contents on Windows:

```
?rwxrwxrwx 0/0    1018368 2024-10-15 08:55:54 Library/bin/mamba.exe 
?rwxrwxrwx 0/0        137 2024-10-15 08:55:54 Scripts/activate.bat 
?rwxrwxrwx 0/0       1911 2024-10-15 08:55:54 condabin/_mamba_activate.bat 
?rwxrwxrwx 0/0        137 2024-10-15 08:55:54 condabin/activate.bat 
?rwxrwxrwx 0/0        874 2024-10-15 08:55:54 condabin/mamba.bat 
?rwxrwxrwx 0/0        791 2024-10-15 08:55:54 condabin/mamba_hook.bat 
```

**Review by JohanMabille — APPROVED** [review]

**Comment by JohanMabille** [comment]
@jaimergp sorry for the long time before coming bac to this!

**Pull request merged**